Fix e2e test for secret command

dustmop opened the pull request:
### What does this PR do?

Fix the e2e test for secret command:

The entry in the test used to be `host_aliases`, but now because slice indices show up, it is `host_aliases/0`.

### Motivation


### Additional Notes



### Possible Drawbacks / Trade-offs


### Describe how to test/QA your changes

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

hush-hush commented on test/new-e2e/tests/agent-subcommands/secret/secret_nix_test.go:
```diff
@@ -63,7 +63,7 @@ host_aliases:
 	assert.Contains(v.T(), output, "Group: dd-agent")
 	assert.Regexp(v.T(), "Number of secrets .+: 1", output)
-	assert.Contains(v.T(), output, "- 'alias_secret':\n\tused in 'datadog.yaml' configuration in entry 'host_aliases'")
+	assert.Contains(v.T(), output, "- 'alias_secret':\n\tused in 'datadog.yaml' configuration in entry 'host_aliases")
```
This feels like the code is missing a quote no ? Should we fix it rather than the test ?

dustmop replied:
That was in order to match either `'host_aliases'` or `'host_aliases/0'`, the common prefix is `'host_aliases` (missing final quote). However I found the proper fix instead, so now it's just using `'host_aliases/0'`.

dustmop committed:
Fix the output of `agent secret`.

Output of this command was changed by https://github.com/DataDog/datadog-agent/pull/21600 by including slice indexes in config paths for secrets.

dustmop commented:
Was having trouble getting this test to pass locally, because the e2e framework would use the most recently deployed agent, hence the original hacky solution. However, I learned about the `E2E_PIPELINE_ID` envvar to use a CI pipeline build instead, which fixes the problem.
